src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -196,11 +196,12 @@ private async Task<HtmlString> RenderSectionAsyncCore(string sectionName, bool r
             else if (required)
             {
                 // If the section is not found, and it is not optional, throw an error.
-                var message = Resources.FormatSectionNotDefined(
-                    ViewContext.ExecutingFilePath,
-                    sectionName,
-                    ViewContext.View.Path);
-                throw new InvalidOperationException(message);
+                var viewContext = ViewContext;
+                throw new InvalidOperationException(
```

Comment:
This change is moving the lookup for the throw to the right hand-side of the throw; so the Jit can identify it as "code" code and deprioritize it in the flow (i.e. anything to the right of a throw is called rarely)